converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/TableStorage/TableStoreServiceUnImplementedForFeatureFlag.cs

```diff
@@ -0,0 +1,25 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Health.DicomCast.Core.Features.ExceptionStorage;
+
+namespace Microsoft.Health.DicomCast.Core.Features.TableStorage
+{
+    /// <summary>
+    /// Implementation of ITableStoreService to use when the feature flag is not enabled as the regular
+    /// implementation requires a CloudTableClient which is not initialized when the feature flag is disabled
+    /// </summary>
+    public class TableStoreServiceUnImplementedForFeatureFlag : ITableStoreService
```

Comment:
Rather than having this, I wonder if we should abstract a bit here. Instead of this being an `ITableStoreService` should this be something like an `IExceptionStore`. This implementation could then be something like a `NullExceptionStore` (or something) that essentially no-ops instead of throwing?

Reply:
Refactored, thanks for the help!